components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -645,6 +648,73 @@ private void revokeAccessTokensWhenSaaSDisabled(final ServiceProvider servicePro
         }
     }
 
+    /**
+     * Revokes tokens of OAuth applications if application is disabled.
+     *
+     * @param serviceProvider Service Provider
+     * @param tenantDomain    Application tenant domain
+     */
+    private void revokeTokensWhenApplicationDisabled(ServiceProvider serviceProvider, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        InboundAuthenticationRequestConfig[] configs = serviceProvider.getInboundAuthenticationConfig()
+                .getInboundAuthenticationRequestConfigs();
+        for (InboundAuthenticationRequestConfig config : configs) {
+            if (IdentityApplicationConstants.OAuth2.NAME
+                    .equalsIgnoreCase(config.getInboundAuthType()) &&
+                    config.getInboundAuthKey() != null) {
+                String oauthKey = config.getInboundAuthKey();
+                int countToken = 0;
+                Set<AccessTokenDO> activeDetailedTokens;
+                try {
+                    activeDetailedTokens = OAuthTokenPersistenceFactory
+                            .getInstance().getAccessTokenDAO().getActiveAcessTokenDataByConsumerKey(oauthKey);
+                    String[] accessTokens = new String[activeDetailedTokens.size()];
+                    for (AccessTokenDO detailToken : activeDetailedTokens) {
+                        String token = detailToken.getAccessToken();
+                        accessTokens[countToken] = token;
+                        countToken++;
+                    }
+                    Set<String> authorizationCodes = OAuthTokenPersistenceFactory.getInstance()
```

Comment:
let's have a separate private method for authorization code revokes

Reply:
Resolved with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2484/commits/1f0d6cb338178ec6f1ea45294448c5470696a61f